### ISSUE TITLE
URL query parameters are dropped in request encoding

### ISSUE THREAD

**Issue by tanyav2** [dajiaji/bhttp-js#191]
The current encoder loses URL query parameters by only using `url.pathname` when encoding the request's path component. 
https://github.com/dajiaji/bhttp-js/blob/46667076981acdee6e3f6817ddb602581e59388f/src/encoder.ts#L168

A fix would involve something like the following two pieces:
1. Change path encoding
```typescript
// Current:
this.encodeVliAndValue(ctx, ctx.url.pathname);
// Should be:
this.encodeVliAndValue(ctx, ctx.url.pathname + ctx.url.search);
```

2. Buffer size calculation:
```typescript
// Current:
len += this.calculateVliSize(ctx.url.pathname.length);
len += ctx.url.pathname.length;
// Should be:
const fullPath = ctx.url.pathname + ctx.url.search;
len += this.calculateVliSize(fullPath.length);
len += fullPath.length;
```

**Comment by dajiaji** [comment]
@tanyav2 Thank you for your report! Are you planning to send a PR to fix this issue? If so, I will wait. If not, please let us know, and I'll fix it promptly on my end.

**Comment by tanyav2** [comment]
Hey @dajiaji, please feel free to fix! It'll probably be far quicker for you.

**Comment by dajiaji** [comment]
Okay, please wait for a whie.

**Comment by dajiaji** [comment]
@tanyav2 sorry for the late. I've released 0.3.6, which resolves the issue.

**Comment by tanyav2** [comment]
> @tanyav2 sorry for the late. I've released 0.3.6, which resolves the issue.

Thank you so much for the quick fix, really appreciate it!
